components/modal.jsx

```diff
@@ -0,0 +1,31 @@
+// [THIS LINE TO BE REMOVED] bootstrap class reference: http://nakupanda.github.io/bootstrap3-dialog/
+
+var React = require('react');
+var ReactAddons = require('react/addons');
+
+var Modal = React.createClass({
+  mixins: [ReactAddons.PureRenderMixin],
+  handleClick: function(e) {
+    if (e.target === this.getDOMNode().querySelector("button.close")) {
+      this.props.onClose();
+    }
+  },
+  render: function() {
+    return (
+      <div className="modal show">
+        <div className="modal-backdrop"></div>
+        <div className="modal-dialog" onClick={this.handleClick}>
+          <div className="modal-header">
+            <button type="button" className="close" data-dismiss="modal" aria-hidden="true">x</button>
```

Comment:
Doh, this is a problem w/ my code since I gave you this file, but I just realized that `data-dismiss` shouldn't be in here either, since it's specific to bootstrap's jQuery-based JavaScript stuffs. If it's easy for you to remove then that would be cool, but otherwise I will do it since it's my fault!